integration/cli/build.configequality.test.ts

```diff
@@ -0,0 +1,424 @@
+            {
+              apiGroups: ["test"],
+              resources: ["tests"],
+              verbs: ["get", "list"],
+            },
+          ],
+        };
+        await fs.writeFile(packageJson, JSON.stringify(config, null, 2));
+
+        // translate package.json values into ModuleConfig
+        peprUuid = `pepr-${config.pepr.uuid}`;
+        moduleConfig = config.pepr;
+        moduleConfig.appVersion = config.version;
+        moduleConfig.description = config.description;
+
+        // build module
+        const build = await pepr.cli(testModule, { cmd: `pepr build` });
+        expect(build.exitcode).toBe(0);
+        expect(build.stderr.join("").trim()).toBe("");
+        expect(build.stdout.join("").trim()).toContain("K8s resource for the module saved");
+
+        // load helm manifests
+        const chartDir = `${testModule}/dist/${moduleConfig.uuid}-chart`;
+        const helm = await pepr.cli(chartDir, { cmd: `helm template .` });
+        expect(build.exitcode).toBe(0);
+        expect(build.stderr.join("").trim()).toBe("");
+        expect(build.stdout.join("").trim()).not.toBe("");
+
+        const helmManifest = `${chartDir}/reified.yaml`;
+        await fs.writeFile(helmManifest, helm.stdout.join("\n"));
+        helmResources = await resource.fromFile(helmManifest);
+
+        // load pepr manifests
+        const peprManifest = `${testModule}/dist/pepr-module-${moduleConfig.uuid}.yaml`;
+        peprResources = await resource.fromFile(peprManifest);
+      }, time.toMs("2m"));
+
+      it("peprVersion", async () => {
+        const peprStringified = JSON.stringify(peprResources);
+        const helmStringified = JSON.stringify(helmResources);
+
+        expect(peprStringified.includes(moduleConfig.peprVersion)).toBe(false);
+        expect(helmStringified.includes(moduleConfig.peprVersion)).toBe(false);
+      });
+
+      it("appVersion", async () => {
+        const peprStringified = JSON.stringify(peprResources);
+        const helmStringified = JSON.stringify(helmResources);
+
+        expect(peprStringified.includes(moduleConfig.appVersion)).toBe(false);
+        expect(helmStringified.includes(moduleConfig.appVersion)).toBe(false);
+      });
+
+      // eslint-disable-next-line max-statements, complexity
+      it("uuid", async () => {
+        for (const clusterRole of [
+          resource.select(peprResources, kind.ClusterRole, peprUuid),
+          resource.select(helmResources, kind.ClusterRole, peprUuid),
+        ]) {
+          expect(clusterRole).toBeDefined();
+        }
+
+        for (const clusterRoleBinding of [
+          resource.select(peprResources, kind.ClusterRoleBinding, peprUuid),
+          resource.select(helmResources, kind.ClusterRoleBinding, peprUuid),
+        ]) {
+          expect(clusterRoleBinding).toBeDefined();
+          expect(clusterRoleBinding.roleRef.name).toBe(peprUuid);
+          expect(clusterRoleBinding!.subjects!.at(0)!.name).toBe(peprUuid);
+        }
```

Comment:
Here's an example of what an alternate to the for-loop pattern may look like: 
```suggestion
        [
          resource.select(peprResources, kind.ClusterRoleBinding, peprUuid),
          resource.select(helmResources, kind.ClusterRoleBinding, peprUuid),
        ].flat().forEach(clusterRoleBinding => {
          expect(clusterRoleBinding).toBeDefined();
          expect(clusterRoleBinding.roleRef.name).toBe(peprUuid);
          expect(clusterRoleBinding!.subjects!.at(0)!.name).toBe(peprUuid);
        });
```

Reply:
```
We can use flatMap(), map(), or forEach() for situations where we're iterating over collections.
```
That's true.  But we could also just use plain-ole loops, right?  In this case, there's nothing _wrong_ with the classic loops, and I chose it intentionally since I _liked_ that the loop variable was in the top line.  IMO,  it increases scannability of the file & helps set context for how a dev is intended to read the contents of the loop.

For example:
```js
for (const deployAdmission of [ 
  resource.select(peprResources, kind.Deployment, peprUuid),
  resource.select(helmResources, kind.Deployment, peprUuid),
]) {
  expect(deployAdmission.metadata!.annotations!["pepr.dev/description"]).toBe(
    moduleConfig.description,
  );
}
```
That top line:
```
for (const deployAdmission of [ 
```
helps hint that what's coming next are Deployments of the _admission_ flavor, for example.  It's a way to use my loop variable naming to help top-down readability.

Burying that variable between the loop definition and the functional function body just loses out on an opportunity & increases reading friction, IMO.  I read the array of values and think, "What are these Deployments of?" and then I have to go digging for an answer.

Also, just more generally, I find plain ole' loops tend to be more resilient against oft-unintended consequences of mixing .forEach-like iteration with the async/await-style programming.  Simple-ish S.O explanation here: https://stackoverflow.com/a/37576787 (in short, there be dragons).

Given all that, I think I'd prefer to keep the for loops here.

Reply:
Sure thing, that's certainly more of a stylistic point.  Appreciate the POV